base/workerpool.jl

```diff
@@ -0,0 +1,85 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+
+type WorkerPool
+    channel::RemoteChannel{Channel{Int}}
+end
+
+
+isbusy(pool::WorkerPool) = !isready(pool.channel)
+
+
+"""
+    WorkerPool(workers)
+
+Create a WorkerPool from a vector of worker ids.
+"""
+function WorkerPool(workers::Vector{Int})
+
+    # Create a shared queue of workers...
+    pool = RemoteChannel(()->Channel{Int}(128)) 
+
+    # Check that workers are not already part of a pool...
+    check = () -> if :_worker_pool in names(Main)
+        error("Worker $(myid()) already in a WorkerPool!")
+    end
+    foreach(fetch, [@spawnat w check() for w in workers])
+
+    # Put each worker into the pool...
+    for w in workers
+        put!(pool, w)
+        @spawnat w global _worker_pool = pool
+    end
+
+    WorkerPool(pool)
+end
+
+
+"""
+    WorkerPool(n)
+
+Launches `n` workers using the in-built LocalManager.
+"""
+WorkerPool(n::Integer) = WorkerPool(addprocs(n))
+
+
+"""
+    WorkerPool()
+
+Equivalent to WorkerPool(CPU_CORES)
+"""
+WorkerPool() = WorkerPool(addprocs())
+
+
+"""
+    remotecall_fetch(f, pool::WorkerPool, args...)
+
+Call `f(args...)` on one of the workers in `pool`.
+"""
+function remotecall_fetch(f, pool::WorkerPool, args...)
+    l = (args...)->try f(args...) finally put!(_worker_pool, myid()) end
+    remotecall_fetch(l, take!(pool.channel), args...)
+end
+
+
+"""
+    default_worker_pool()
+
+Built-in WorkerPool (used by `remote(f)`).
+"""
+global _default_worker_pool = nothing
+function default_worker_pool()
```

Comment:
As a first step, I've changed `default_worker_pool` to be empty by default.
For now, you can do `append!(default_worker_pool(), workers())` after calling `addprocs()` or passing `-p` on the command line.

A future PR could automatically call `push!(default_worker_pool(), w)` from `addprocs`. At this point I haven't got my head around the initialisation ordering issues that might crop up...